### PR TITLE
Create Attendance list

### PR DESCRIPTION
**Pull request by ghulamhafeez**
### Implementation

* Create Attendance list and show all attendance 

I've added temporary table action, but will create a separate ticket for this soon. also create ticket for DateRange filter to the Attendance table, set to last 30 days, and a CommonMemberSelector filter to the Attendance table.

### UI

image

### CODE REVIEW

File: src/app/attendance/actions/getAttendance.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+"use server";
+
+import { getSupabaseClient } from "../../../../utils/supabase/supabaseClient";
```
Comment (Monfernape) [inline src/app/attendance/actions/getAttendance.ts:3]: Why relative path?

File: src/app/attendance/actions/getAttendance.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+export const getAttendace = async () => {
+  const supabase = await getSupabaseClient();
+  const { data, error } = await supabase.from("attendance").select("*");
```
Comment (Monfernape) [inline src/app/attendance/actions/getAttendance.ts:7]: Can we get this from enum defined for tables?

File: src/app/attendance/actions/getAttendance.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    return {
+      success: false,
+      message: "There are no attendance available at this time.",
+      data: [],
+    };
+  }
+
+  return {
+    success: true,
+    message: "Attendance loaded successfully.",
```
Comment (Monfernape) [inline src/app/attendance/actions/getAttendance.ts:19]:
```suggestion
    message: "Attendances loaded successfully.",
```

File: src/app/attendance/components/AttendanceList.tsx [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+    return member && member.shift ? member.shift : "Unknown Shift";
+  };
+
+  const columns: ColumnDef<Attendance>[] = [
+    {
+      accessorKey: "memberId",
+      header: "Name",
+      cell: ({ row }) => (
+        <div className="capitalize">
+          {getNameById(row.getValue("memberId"))}
```
Comment (Monfernape) [inline src/app/attendance/components/AttendanceList.tsx:63]: Maybe we don't to pass members as prop. I think we can map over that in the server component and get required result there. Wdyt?

File: src/app/attendance/components/AttendanceList.tsx [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+    {
+      accessorKey: "status",
+      header: "Status",
+      cell: ({ row }) => <UserStatusBadge status={row.getValue("status")} />,
+    },
+    {
+      id: "actions",
+      enableHiding: false,
+      header: () => {
+        return <span>Action</span>;
```
Comment (Monfernape) [inline src/app/attendance/components/AttendanceList.tsx:96]: Why span here?

File: src/app/attendance/components/AttendanceList.tsx [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    return member ? member.name : "Unknown";
+  };
+
+  const getShiftByMemberId = (memberId: number) => {
+    const member = memberData.find((member) => member.id === memberId);
+
+    return member && member.shift ? member.shift : "Unknown Shift";
+  };
+
+  const columns: ColumnDef<Attendance>[] = [
```
Comment (Monfernape) [inline src/app/attendance/components/AttendanceList.tsx:57]: Columns should be memoized. Read tanstack table docs.

File: src/app/attendance/components/AttendanceList.tsx [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+  members: {
+    data: Member[];
+    success: boolean;
+    message: string;
+  };
+}
+
+export const AttendanceList = ({ attendance, members }: AttendanceProps) => {
+  const { toast } = useToast();
+  const { data: attendanceData, success } = attendance;
```
Comment (Monfernape) [inline src/app/attendance/components/AttendanceList.tsx:42]: Why do we need `success` here?

File: src/app/attendance/components/AttendanceList.tsx [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+  useEffect(() => {
+    if (!success) {
+      toast({
+        title: "No Attendance Found",
+        description: "There are no members available at this time.",
+      });
+    }
+  }, [success, toast]);
+
+  const toNavigate = () => {
```
Comment (Monfernape) [inline src/app/attendance/components/AttendanceList.tsx:119]: This should be named better.

File: src/app/attendance/page.tsx [hunk 9]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { PageLayout } from "../layout/PageLayout";
+import { AttendanceList } from "./components/AttendanceList";
+import { getAttendace } from "./actions/getAttendance";
+import { getMembers } from "../members/actions/getMembers";
+
+const Page = async () => {
+  const attendance = await getAttendace();
+  const members = await getMembers();
```
Comment (Monfernape) [inline src/app/attendance/page.tsx:9]: Maybe we can remove `members` as a prop and instead we can just map over attendances and map users here. Wdyt?

File: src/app/attendance/page.tsx [hunk 10]
```diff
@@ -0,0 +1,19 @@
+import React from "react";
+import { PageLayout } from "../layout/PageLayout";
+import { AttendanceList } from "./components/AttendanceList";
+import { getAttendace } from "./actions/getAttendance";
+import { getMembers } from "../members/actions/getMembers";
+
+const Page = async () => {
+  const attendance = await getAttendace();
```
Comment (Monfernape) [inline src/app/attendance/page.tsx:8]: Typo

File: src/types/index.tsx [hunk 11]
```diff
@@ -9,7 +9,13 @@ export interface Member {
   role: UserRole;
   status: UserStatus;
 }
-
+export interface Attendance {
```
Comment (Monfernape) [inline src/types/index.tsx:12]: Why is this here? This should be localized?

**Review by Monfernape — CHANGES_REQUESTED** [review]

File: src/app/members/components/MemberActionRender.tsx [hunk 12]
```diff
@@ -1,13 +1,13 @@
-import React from 'react';
-import ActionsMenu from '@/common/ActionMenu/ActionsMenu';
+import React from "react";
+import ActionsMenu from "@/common/ActionMenu/ActionsMenu";
 import { Info, Trash2, Edit, Send } from "lucide-react";
-import { MemberRole } from '@/lib/constants';
+import { MemberRole } from "@/lib/constants";
 
 interface MemberInfo {
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/members/components/MemberActionRender.tsx:6]: @ghulamhafeez is here need to create seprate interface?

File: src/app/attendance/page.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+  const members = await getMembers();
+  return (
+    <>
```
Comment (muhammad-bilal-ali-khokhar) [inline src/app/attendance/page.tsx:11]: @ghulamhafeez please remove this fragments

File: src/app/members/components/MemberActionRender.tsx [hunk 14]
```diff
@@ -1,13 +1,13 @@
...
   id: number;
-  name: string;
-  phone: string;
-  role: string;
+  name?: string;
+  phone?: string;
+  role?: string;
```
Comment (nadirhussain786) [inline src/app/members/components/MemberActionRender.tsx:10]: @ghulamhafeez why it need to be optional?

File: src/app/attendance/components/AttendanceList.tsx (hunk 5 above)
Comment (ghulamhafeez) [reply src/app/attendance/components/AttendanceList.tsx:96]: Fixed

File: src/app/attendance/components/AttendanceList.tsx (hunk 7 above)
Comment (ghulamhafeez) [reply src/app/attendance/components/AttendanceList.tsx:42]: Fixed. I initially used an older approach, but now I've updated it to use the latest method.

File: src/app/attendance/components/AttendanceList.tsx (hunk 8 above)
Comment (ghulamhafeez) [reply src/app/attendance/components/AttendanceList.tsx:119]: Fixed

File: src/app/members/components/MemberActionRender.tsx (hunk 14 above)
Comment (ghulamhafeez) [reply src/app/members/components/MemberActionRender.tsx:10]: For the time being, we’re using a workaround. I’ve also mentioned this in the PR, and we’ve created a separate ticket to address it
